src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/DoNotUseCompilationGetSemanticModelTests.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.DoNotUseCompilationGetSemanticModelAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.DoNotUseCompilationGetSemanticModelAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.CodeAnalysis.Analyzers.UnitTests.MetaAnalyzers
+{
+    public class DoNotUseCompilationGetSemanticModelTests
+    {
+        [Fact]
+        public async Task CallInInitialize()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+class Analyzer : DiagnosticAnalyzer
+{
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => throw null;
+
+    public override void Initialize(AnalysisContext context)
+    {
+        context.RegisterCompilationStartAction(csac =>
+        {
+            csac.Compilation.GetSemanticModel(null);
```

Comment:
Can you add a test where this API is invoked outside a DiagnosticAnalyzer and ensure we don't flag it?